### PR TITLE
Premium Content Block: restrict registration to wpcom

### LINKED ISSUE #18521: Premium Content Block: default Payment plan does not exist when you first add a block
#### Steps to reproduce the issue

1. Start with a site running `master`, and connected to WordPress.com.
2. Purchase a plan giving you access to the Premium Content block (e.g. Complete).
3. Add the block to a new post.
4. Follow the prompt to connect your site to Stripe.
5. Once back in the editor, publish your post.
6. You will not see a payment button on the frontend, only in the editor.
7. Add a separate Payment button outside of the Premium Content block, create a new Payment plan inside the Payment block controls, and update the post.
8. Now that a Payment plan exists, the Payment button can appear on the frontend, both in the Premium Content block and in the separate Payment block.

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #18521

#### Changes proposed in this Pull Request:

We already do the same on the PHP side; let's also limit the use of the block to WordPress.com sites (simple and atomic).

**Note**: by doing so, we would break the block for anyone using it on their self-hosted site today by using the experimental or beta flags. However, I should note that we intended for the block to only be available on WordPress.com from the start (see d80d3d96cd3c762007e5f67b4a802c955a796621 and c39578af43b89097c53f38ff79d62ce960775321 in the PR that introduced the feature).

#### Jetpack product discussion

* See https://github.com/Automattic/jetpack/issues/18521#issuecomment-1031659152

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* On a Jetpack site that's connected to WordPress.com, and with a complete plan, enable Beta blocks: `define( 'JETPACK_BETA_BLOCKS', true );` in `wp-config.php`.
* Go to Posts > Add New
* Search for the Premium Content block; it should not be offered to you.
* On a WoA site (or a wpcom simple site with a paid plan running D74536-code), the block should still be available.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/premium-content/editor.js [hunk 1]
```diff
@@ -2,29 +2,33 @@
...
 } from './subscriber-view/.';
-import {
-	name as buttonsBlockName,
-	settings as buttonsBlockSettings
-} from './buttons/.';
+import { name as buttonsBlockName, settings as buttonsBlockSettings } from './buttons/.';
 import {
 	name as loginButtonBlockName,
-	settings as loginButtonBlockSettings
+	settings as loginButtonBlockSettings,
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/premium-content/editor.js:18]: Those were automatic updates by Prettier.

**Review by Copons — APPROVED** [review]

**Review by kraftbj — DISMISSED** [review]

**Comment by kraftbj** [comment]
I approved it, but there's a failure when building on WP.com:

```
[build-extensions] ERROR in ./extensions/blocks/premium-content/editor.js 17:4-16
[23:17:34]
[build-extensions] export 'isSimpleSite' (imported as 'isSimpleSite') was not found in '@automattic/jetpack-shared-extension-utils' (possible exports: JETPACK_DATA_PATH, getJetpackData, getJetpackExtensionAvailability, registerJetpackPlugin)
```

**Comment by jeherve** [comment]
It needed a rebase on WordPress.com, it's okay now.

**Comment by jeherve** [comment]
> It require merge from master to resolve conflicts.

👍 Done.

**Review by brbrr — APPROVED** [review]
Tested well in Jetpack and AT sites

**Comment by jeherve** [comment]
r240028-wpcom

**Pull request merged**